backend/app/models/schedule.py

```diff
@@ -31,3 +40,47 @@ class SchedulePublic(BaseModel):
     schedule: ScheduleInfo
     id: int | None
     scheduled_inquiries: list[int]
+
+    @computed_field
+    def scheduled_inquiries_and_dates(self) -> ScheduleInquiriesAndDates:
+        schedule = self.schedule
+        scheduled_inquiries = self.scheduled_inquiries
+        today = datetime.strptime(
+            f"{datetime.now().strftime('%Y-%m-%d')} {schedule.timesOfDay[0]}",
+            "%Y-%m-%d %H:%M",
+        )
+        start = datetime.strptime(
+            f"{schedule.startDate} {schedule.timesOfDay[0]}", "%Y-%m-%d %H:%M"
+        )
+        current_date = start
+        past_scheduled_count = 0
+        while current_date < today:
+            if schedule.skipWeekends:
+                while current_date.weekday() in [5, 6]:  # 5: Saturday, 6: Sunday
+                    current_date += timedelta(days=1)
+            if schedule.skipHolidays:
+                us_holidays = holidays.US()  # type: ignore[attr-defined]
+                while current_date in us_holidays:  # Adjust country_code as needed
+                    current_date += timedelta(days=1)
+            current_date += timedelta(days=schedule.daysBetween)
+            past_scheduled_count += 1
+        scheduled_inquiries_count = len(scheduled_inquiries)
+        scheduled_dates: list[str] = []
+        for _index in scheduled_inquiries:
+            if schedule.skipWeekends:
```

Comment:
given the repeated set of checks, might consider a `skip` function that returns the number of days that should be skipped.